pymc_marketing/prior.py

```diff
@@ -1025,8 +1026,263 @@ def create_likelihood_variable(
         return distribution.create_variable(name)
 
 
+class VariableNotFound(Exception):
+    """Variable is not found."""
+
+
+def _remove_random_variable(var: pt.TensorVariable) -> None:
+    if var.name is None:
+        raise ValueError("This isn't removable")
+
+    name: str = var.name
+
+    model = pm.modelcontext(None)
+    for idx, free_rv in enumerate(model.free_RVs):
+        if var == free_rv:
+            index_to_remove = idx
+            break
+    else:
+        raise VariableNotFound(f"Variable {var.name!r} not found")
+
+    var.name = None
+    model.free_RVs.pop(index_to_remove)
+    model.named_vars.pop(name)
+
+
+@dataclass
```

Comment:
What about https://docs.pydantic.dev/latest/api/base_model/ ? or even or even https://docs.pydantic.dev/latest/concepts/dataclasses/ for data validation as we are using it on other parts of the code?

Reply:
Switched in fe3d70c5